### PR TITLE
fix a tz bug when validate ErrTooHighTimestampPrecision

### PR DESCRIPTION
**Pull request by Yifeng-Sigma**
### Description
We got error `268003 (22007): Cannot convert timestamp 2021-01-01 01:24:54 +0000 UTC in column Date to Arrow.Timestamp data type due to too high precision. Please use context with WithOriginalTimestamp.` However this should be within the range that nanosecond supports. 

Notice that the date is the first day of the year, that points to where the problem is. The right side is always in UTC while the left side is from `arrowSnowflakeTimestampToTime()` and can have a timezone. 

### Checklist
- [ ] Code compiles correctly
- [ ] Run ``make fmt`` to fix inconsistent formats
- [ ] Run ``make lint`` to get lint errors and fix all of them
- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: converter.go [hunk 1]
```diff
@@ -1084,7 +1084,7 @@ func arrowToRecord(ctx context.Context, record arrow.Record, pool memory.Allocat
 							ar = arrow.Timestamp(ts.UnixNano())
 							// in case of overflow in arrow timestamp return error
 							// this could only happen for nanosecond case
-							if ts.Year() != ar.ToTime(arrow.Nanosecond).Year() {
+							if ts.UTC().Year() != ar.ToTime(arrow.Nanosecond).Year() {
```
Comment (sfc-gh-pfus) [inline converter.go:1087]: Great catch indeed! Can you add a test to confirm it doesn't work without the fix and works with it?

Comment (Yifeng-Sigma) [reply converter.go:1087]: @sfc-gh-pfus done, using the existing test case, but changed to 1/1. The test failed before the fix.

**Review by sfc-gh-pfus — APPROVED** [review]

**Comment by Yifeng-Sigma** [comment]
@sfc-gh-pfus could you please re-trigger the tests? I think they might be flaky?

**Comment by sfc-gh-dheyman-1** [comment]
Hi @Yifeng-Sigma, I rerun the tests and it seems that everything is passing.

**Comment by Yifeng-Sigma** [comment]
> Hi @Yifeng-Sigma, I rerun the tests and it seems that everything is passing.

@sfc-gh-dheyman-1 Thank you! can we merge this?

**Comment by sfc-gh-dheyman-1** [comment]
Thank you @Yifeng-Sigma for fixing this!

**Pull request merged**